### PR TITLE
300 placeholder intrinsics

### PR DESCRIPTION
**Pull request by CblPOK-git**

### CODE REVIEW

File: include/nil/blueprint/extract_constructor_parameters.hpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+#ifndef CRYPTO3_ASSIGNER_EXTRACT_CONSTRUCTOR_PARAMETERS_HPP
+#define CRYPTO3_ASSIGNER_EXTRACT_CONSTRUCTOR_PARAMETERS_HPP
+
+namespace nil {
+    namespace blueprint {
+        namespace detail {
+
+            template <typename BlueprintFieldType>
+            typename BlueprintFieldType::value_type extract_component_constructor_parameter_field(llvm::Value *input_value){
```
Comment (makxenov) [inline include/nil/blueprint/extract_constructor_parameters.hpp:34]: As I see, the functions in this file are not specific to `component`, `constructor`, `parameter`, or `intrinsic`. They just extract values from a constant. Maybe rename them to something like `extract_constant_field_value` and so on?

File: include/nil/blueprint/parser.hpp [hunk 2]
```diff
@@ -87,9 +87,38 @@
...
+                        ASSERT(constants_positions[j] < inst->getNumOperands() - 1);
+                        is_const = true;
+                        if(!llvm::isa<llvm::Constant>(inst->getOperand(i))) {
+                            std::cerr << "\noperand " << i << " must be constant, but it is not\n";
+                            return false;
+                        }
+                    }
+                }
+                if(!is_const && llvm::isa<llvm::Constant>(inst->getOperand(i))) {
+                    std::cerr << "\noperand " << i << " must NOT be constant, but it is\n";
```
Comment (makxenov) [inline include/nil/blueprint/parser.hpp:115]: Why couldn't an arbitrary argument be a constant?

File: include/nil/blueprint/parser.hpp [hunk 3]
```diff
@@ -510,6 +542,51 @@ namespace nil {
...
+                        var y0 = frame.scalars[inst->getOperand(1)];
+                        var y1 = frame.scalars[inst->getOperand(2)];
+                        var alpha = frame.scalars[inst->getOperand(3)];
+
+                        using component_type = components::fri_lin_inter<
+                            crypto3::zk::snark::plonk_constraint_system<BlueprintFieldType, ArithmetizationParams>, BlueprintFieldType>;
+
+                        component_type component_instance({0, 1, 2, 3, 4}, {}, {});
+
+                        components::generate_circuit(component_instance, circuits[currProverIdx], assignments[currProverIdx], {s, y0, y1, alpha}, start_row);
```
Comment (makxenov) [inline include/nil/blueprint/parser.hpp:556]: Let's move circuit generation to separate handler to conform with other components.

File: include/nil/blueprint/parser.hpp [hunk 4]
```diff
@@ -1234,9 +1311,8 @@ namespace nil {
                                     } else if (ret_val->getType()->isVectorTy()) {
                                         std::vector<var> res = extracted_frame.vectors[ret_val];
                                         for (var x : res) {
-                                            std::cout << var_value(assignments[currProverIdx], x).data << " ";
+                                            std::cout << var_value(assignments[currProverIdx], x).data << "\n";
```
Comment (makxenov) [inline include/nil/blueprint/parser.hpp:1314]: `std::endl` is preferred for cout, it flushes stdout buffer.

**Review by makxenov — CHANGES_REQUESTED** [review]

File: include/nil/blueprint/extract_constructor_parameters.hpp (hunk 1 above)
Comment (CblPOK-git) [reply include/nil/blueprint/extract_constructor_parameters.hpp:34]: This is constants that are component constructor parameters and must not be written into assignment table. But also we have constant column, maybe in the future we will work with the second type of constants field values and confusion may occur. This way of naming distinguishes two cases

File: include/nil/blueprint/parser.hpp (hunk 2 above)
Comment (CblPOK-git) [reply include/nil/blueprint/parser.hpp:115]: We don't put constants into assignment table, but components need them in the table

File: include/nil/blueprint/parser.hpp (hunk 3 above)
Comment (CblPOK-git) [reply include/nil/blueprint/parser.hpp:556]: ok

File: include/nil/blueprint/parser.hpp (hunk 4 above)
Comment (CblPOK-git) [reply include/nil/blueprint/parser.hpp:1314]: ok

File: include/nil/blueprint/parser.hpp (hunk 2 above)
Comment (makxenov) [reply include/nil/blueprint/parser.hpp:115]: Well, let's put them into the table in this case

Comment (CblPOK-git) [reply include/nil/blueprint/parser.hpp:115]: The whole idea was to use constants as a way to pass variables to intrinsic without writing them into table. Some args must be in the table ( =must not be constant). Some args must not be in the table ( =must be constant).  

For example, this is okay:
`var input;
const int parameter;
__some_intrinsic(input, parameter);`

And this two variants are not okay and will cause problems:
`const var input;
const int parameter;
__some_intrinsic(input, parameter);`
`const var input;
int parameter;
__some_intrinsic(input, parameter);`

I check that every argument that must be constant is constant and vice versa.

File: include/nil/blueprint/parser.hpp [hunk 5]
```diff
@@ -116,6 +123,28 @@ namespace nil {
...
+                    is_const = false;
+                    for (std::size_t j = 0; j < constants_positions.size(); j++) {
+                        if(i == constants_positions[j]) {
+                            ASSERT(constants_positions[j] < inst->getNumOperands() - 1);
+                            is_const = true;
+                            if(!llvm::isa<llvm::Constant>(inst->getOperand(i))) {
+                                std::cerr << "\noperand " << i << " must be constant, but it is not\n";
+                                return false;
+                            }
+                        }
```
Comment (makxenov) [inline include/nil/blueprint/parser.hpp:138]: Maybe `break` here?

**Review by makxenov — APPROVED** [review]

**Pull request merged**